### PR TITLE
[Feature][Jdbc] Support read multiple tables

### LINKED ISSUE #5482: [Feature][Connector] JDBC source supports multi-table reading in one task
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Supports multi-table reading in one task to save resource and speed up.

STIP: https://cwiki.apache.org/confluence/display/SEATUNNEL/STIP4-JDBC+source+supports+multi-table+reading+in+one+task

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #5675: [Feature][Jdbc] Support auto split for jdbc source
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Support auto split for jdbc source

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

close #5482

Support read table list for jdbc connector

### Does this PR introduce _any_ user-facing change?

Compatible with older versions config：
```
source {
    Jdbc {
        url = "jdbc:mysql://127.0.0.1:3306/test"
        driver = "com.mysql.cj.jdbc.Driver"
        connection_check_timeout_sec = 100
        user = "root"
        password = "root@123"

        query = "select * from test_1"
        partition_column = "id"
        partition_num = 100
    }
}
```

Support new config of table path:
```
source {
    Jdbc {
        url = "jdbc:mysql://127.0.0.1:3306/test"
        driver = "com.mysql.cj.jdbc.Driver"
        connection_check_timeout_sec = 100
        user = "root"
        password = "root@123"

        table_path = "test.test_1"

        #split.size = 8096
        #split.even-distribution.factor.upper-bound = 100
        #split.even-distribution.factor.lower-bound = 0.05
        #split.sample-sharding.threshold = 1000
        #split.inverse-sampling.rate = 1000
    }
}
```

Support read table list config:
```
source {
    Jdbc {
        url = "jdbc:mysql://127.0.0.1:3306/test"
        driver = "com.mysql.cj.jdbc.Driver"
        connection_check_timeout_sec = 100
        user = "root"
        password = "root@123"

        table_list = [
            {
                table_path = "test.test_1"
            },
            {
                table_path = "test.test_2"
            },
            {
                query = "select id, name from test.test_3 where id > 1000"
                table_path = "test.test_3"
            }
        ]
        #split.size = 8096
        #split.even-distribution.factor.upper-bound = 100
        #split.even-distribution.factor.lower-bound = 0.05
        #split.sample-sharding.threshold = 1000
        #split.inverse-sampling.rate = 1000
    }
}
```

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hailin0** [comment]
Waiting for #5595、#5604、#5620

File: docs/en/concept/connector-v2-features.md [hunk 1]
```diff
@@ -49,6 +49,10 @@ In the **Parallelism Source Connector**, the source will be split into multiple
 
 User can config the split rule.
 
+### support multiple table reading
```
Comment (Hisoka-X) [inline docs/en/concept/connector-v2-features.md]:
```suggestion
### Support multiple table reading
```

File: docs/en/concept/connector-v2-features.md [hunk 2]
```diff
@@ -49,6 +49,10 @@ In the **Parallelism Source Connector**, the source will be split into multiple
...
+
+Supports reading multiple tables at one time
```
Comment (Hisoka-X) [inline docs/en/concept/connector-v2-features.md]:
```suggestion
Supports reading multiple tables in one SeaTunnel job
```

File: docs/en/connector-v2/source/Jdbc.md [hunk 3]
```diff
@@ -25,25 +25,33 @@ supports query SQL and can achieve projection effect.
...
+| fetch_size                                 | Int    | No       | 0               |
+| properties                                 | Map    | No       | -               |
+| table_path                                 | String | No       | -               |
+| table_list                                 | Array  | No       | -               |
+| split.size                                 | Int    | No       | 8096            |
+| split.even-distribution.factor.lower-bound | Double | No       | 0.05            |
+| split.even-distribution.factor.upper-bound | Double | No       | 100             |
+| split.sample-sharding.threshold            | Int    | No       | 1000            |
+| split.inverse-sampling.rate                | Int    | No       | 1000            |
+| common-options                             |        | No       | -               |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/Jdbc.md:55]: This is a new feature or just document missed? I think it's better create a single PR for this.

File: docs/en/connector-v2/source/Jdbc.md [hunk 4]
```diff
@@ -98,6 +106,54 @@ improve performance by reducing the number database hits required to satisfy the
...
+table_list = [
+  {
+    table_path = "testdb.table1"
+  }
+  {
+    table_path = "testdb.table2"
+    query = "select * from testdb.table2 where id > 100"
+  }
+]
+```
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/Jdbc.md:136]: If we use multiple tables feature in jdbc source, how to deal with config only for single table like: `partition_column`, `partition_upper_bound` etc? I think you should share the behavior to the doc which when use multiple tables.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/dm/DamengCatalog.java [hunk 5]
```diff
@@ -148,4 +152,23 @@ protected String getUrlFromDatabaseName(String databaseName) {
...
+            if (StringUtils.isNotBlank(tablePath.getDatabaseName())) {
+                return databaseExists(tablePath.getDatabaseName())
+                        && listTables(tablePath.getDatabaseName())
+                                .contains(tablePath.getSchemaAndTableName());
+            }
+            return listTables().contains(tablePath.getSchemaAndTableName());
+        } catch (DatabaseNotExistException e) {
+            return false;
+        }
+    }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/dm/DamengCatalog.java:171]: Shoud care about case?

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oracle/OracleCatalog.java [hunk 6]
```diff
@@ -223,4 +231,41 @@ protected String getUrlFromDatabaseName(String databaseName) {
...
+            if (StringUtils.isNotBlank(tablePath.getDatabaseName())) {
+                return databaseExists(tablePath.getDatabaseName())
+                        && listTables(tablePath.getDatabaseName())
+                                .contains(tablePath.getSchemaAndTableName());
+            }
+            return listTables().contains(tablePath.getSchemaAndTableName());
+        } catch (DatabaseNotExistException e) {
+            return false;
+        }
+    }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oracle/OracleCatalog.java:246]: ditto

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcOptions.java [hunk 7]
```diff
@@ -67,7 +67,7 @@ public interface JdbcOptions {
     Option<Integer> FETCH_SIZE =
             Options.key("fetch_size")
                     .intType()
-                    .defaultValue(0)
+                    .defaultValue(1024)
                     .withDescription(
                             "For queries that return a large number of objects, "
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcOptions.java]: Please do not change this value. `0` mean use default value of jdbc driver. Use `1024` as default value could affect some db read performance.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java [hunk 8]
```diff
@@ -61,91 +53,48 @@ public class JdbcSource
...
-                        jdbcSourceConfig.getJdbcConnectionConfig().isAutoCommit());
-    }
-
     @Override
     public Boundedness getBoundedness() {
         return Boundedness.BOUNDED;
     }
 
     @Override
-    public SeaTunnelDataType<SeaTunnelRow> getProducedType() {
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java:141]: Remove `getProducedType` should waiting #5670

File: docs/en/concept/connector-v2-features.md (hunk 1 above)
Comment (hailin0) [reply docs/en/concept/connector-v2-features.md]: I followed the existing codestyle

https://github.com/apache/seatunnel/blob/dev/docs/en/concept/connector-v2-features.md#support-user-defined-split

Comment (Hisoka-X) [reply docs/en/concept/connector-v2-features.md]: Ok, it's fine for now.

File: docs/en/connector-v2/source/Jdbc.md (hunk 3 above)
Comment (hailin0) [reply docs/en/connector-v2/source/Jdbc.md:55]: This feature is tied to multiple tables, and I can create issues linked to this pr

close https://github.com/apache/seatunnel/issues/5675

File: docs/en/connector-v2/source/Jdbc.md (hunk 4 above)
Comment (hailin0) [reply docs/en/connector-v2/source/Jdbc.md:136]: added case into docs example

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcOptions.java (hunk 7 above)
Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/config/JdbcOptions.java]: fixed

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java (hunk 8 above)
Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSource.java:141]: revert

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/dm/DamengCatalog.java (hunk 5 above)
Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/dm/DamengCatalog.java:171]: This is an exist problem and needs to be new issue handle

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/mysql/MySqlCatalog.java [hunk 9]
```diff
@@ -47,7 +50,7 @@ public class MySqlCatalog extends AbstractJdbcCatalog {
     private static final MysqlDataTypeConvertor DATA_TYPE_CONVERTOR = new MysqlDataTypeConvertor();
 
     private static final String SELECT_COLUMNS_SQL_TEMPLATE =
-            "SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = '%s' AND TABLE_NAME ='%s'";
+            "SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = '%s' AND TABLE_NAME ='%s' ORDER BY ORDINAL_POSITION ASC";
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/mysql/MySqlCatalog.java:53]: orderby

**Review by ruanwenjun — DISMISSED** [review]
LGTM

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/ConfigUtil.java [hunk 10]
```diff
@@ -83,6 +83,12 @@ public static <T> T convertValue(Object rawValue, Option<T> option) {
     }
 
     static <T> List<T> convertToList(Object rawValue, Class<T> clazz) {
+        if (rawValue instanceof List) {
+            return ((List<?>) rawValue)
+                    .stream()
+                            .map(value -> convertValue(convertToJsonString(value), clazz))
```
Comment (hailin0) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/ConfigUtil.java:89]: fix bug
```
table_list = [{xxx=xxx},{xxx=xxx}]
```

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java [hunk 11]
```diff
@@ -158,7 +155,10 @@ public CatalogTable getTable(TablePath tablePath)
             throw new TableNotExistException(catalogName, tablePath);
         }
 
-        String dbUrl = getUrlFromDatabaseName(tablePath.getDatabaseName());
+        String dbUrl = getUrlFromDatabaseName(defaultDatabase);
+        if (StringUtils.isNotBlank(tablePath.getDatabaseName())) {
+            dbUrl = getUrlFromDatabaseName(tablePath.getDatabaseName());
+        }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java]:
```suggestion
        String dbUrl;
        if (StringUtils.isNotBlank(tablePath.getDatabaseName())) {
            dbUrl = getUrlFromDatabaseName(tablePath.getDatabaseName());
        } else {
            dbUrl = getUrlFromDatabaseName(defaultDatabase);
        }
```

Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java]: fixed

**Review by Hisoka-X — APPROVED** [review]

**Review by ruanwenjun — APPROVED** [review]
+1

**Pull request merged**